### PR TITLE
Whisper: support 4d mask input for decoder models

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, @ibelem, PTAL, thanks!

File: js/whisper-demo/whisper.js [hunk 1]
```diff
@@ -140,8 +158,20 @@ export class Whisper {
         decoder_input['input_ids'] = new this.ort.Tensor('int32', new Int32Array([new_token]), [1, 1]);
 
         // pad attention mask to max_seq_length
-        decoder_input['attention_mask'] = new this.ort.Tensor('int64', attention_mask_update(this.ort, new BigInt64Array([1n, 1n, 1n, 1n]),
-            0, this.max_sequence_length, this.num_init_tokens), [1, 128]);
+        const mask_data = attention_mask_update(
+            this.ort,
+            this.mask_4d ? new Uint16Array(4) : new BigInt64Array(4).fill(1n),
```
Comment (huningxin) [inline js/whisper-demo/whisper.js]: Nit: explicitly set `0`? `new Uint16Array(4).fill(0)`

**Review by huningxin — APPROVED** [review]
LGTM with a nit

File: js/whisper-demo/whisper.js (hunk 1 above)
Comment (Honry) [reply js/whisper-demo/whisper.js]: OK. Done.

**Comment by Honry** [comment]
Thanks @huningxin, merging.
